integration/setup/common.go

```diff
@@ -104,123 +104,61 @@ func SkipForPostgresWithReason(tb testing.TB, reason string) {
 	}
 }
 
-// checkMongoDBURI returns true if given MongoDB URI is working.
-func checkMongoDBURI(tb testing.TB, ctx context.Context, uri string) bool {
-	tb.Helper()
-
-	defer trace.StartRegion(ctx, "checkMongoDBURI").End()
-	trace.Log(ctx, "checkMongoDBURI", uri)
-
-	clientOpts := options.Client().ApplyURI(uri)
-
-	if *targetTLSF {
-		clientOpts.SetTLSConfig(GetClientTLSConfig(tb))
-	}
-
-	client, err := mongo.Connect(ctx, clientOpts)
-
-	if err == nil {
-		defer client.Disconnect(ctx)
-
-		_, err = client.ListDatabases(ctx, bson.D{})
-	}
-
-	if err != nil {
-		tb.Logf("checkMongoDBURI: %s: %s", uri, err)
-
-		return false
-	}
-
-	tb.Logf("checkMongoDBURI: %s: connected", uri)
-
-	return true
-}
-
 // buildMongoDBURIOpts represents buildMongoDBURI's options.
 type buildMongoDBURIOpts struct {
-	hostPort       string
+	host           string
 	unixSocketPath string
 	tls            bool
+	authMechanism  string
+	user           *url.Userinfo
 }
 
-// buildMongoDBURI builds MongoDB URI with given URI options and validates that it works.
-//
-// TODO rework or remove this https://github.com/FerretDB/FerretDB/issues/1568
-func buildMongoDBURI(tb testing.TB, ctx context.Context, opts *buildMongoDBURIOpts) string {
-	tb.Helper()
+// buildMongoDBURI builds MongoDB URI with given URI options.
+func buildMongoDBURI(tb testing.TB, opts *buildMongoDBURIOpts) string {
+	q := make(url.Values)
 
-	var host string
+	if opts.tls {
+		require.Empty(tb, opts.unixSocketPath, "unixSocketPath cannot be used with TLS")
+		q.Set("tls", "true")
```

Comment:
Now setting tls true, because it's being set in https://github.com/FerretDB/FerretDB/blob/main/Taskfile.yml#L389